File: _inspect_excel.py
import openpyxl, sys, io

sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')

wb = openpyxl.load_workbook('ملف الخريجين القدامى -البصرة 2022.xlsx')
ws = wb.active

rows = list(ws.iter_rows(values_only=True))
print(f'Total rows: {len(rows)}')
print('=== Header Row ===')
for i, h in enumerate(rows[0]):
    print(f'  [{i}] {h}')

print('\n=== First 5 data rows ===')
for r in range(1, min(6, len(rows))):
    vals = []
    for i, v in enumerate(rows[r]):
        if v is not None:
            vals.append(f'[{i}]={repr(v)}')
    print(f'Row {r+1}: {", ".join(vals)}')

File: _inspect_excel.php
<?php
require 'vendor/autoload.php';
use PhpOffice\PhpSpreadsheet\IOFactory;

$file = 'ملف الخريجين القدامى -البصرة 2022.xlsx';
$spreadsheet = IOFactory::load($file);
$sheet = $spreadsheet->getActiveSheet();
$rows = $sheet->toArray();

echo 'Total rows: ' . count($rows) . "\n";
echo "=== Header Row ===\n";
foreach ($rows[0] as $i => $h) {
    echo "[$i] " . (is_null($h) ? 'NULL' : $h) . "\n";
}

echo "=== First 3 data rows ===\n";
for ($r = 1; $r <= min(3, count($rows)-1); $r++) {
    echo 'Row ' . ($r+1) . ": ";
    foreach ($rows[$r] as $i => $v) {
        $val = is_null($v) ? 'NULL' : (is_string($v) ? '"' . $v . '"' : $v);
        echo "[$i]=$val ";
    }
    echo "\n";
}
